applications/FemToDemApplication/custom_processes/fix_free_velocity_on_nodes_process.cpp

```diff
@@ -38,10 +39,45 @@ void FixFreeVelocityOnNodesProcess::Execute()
             it_node->Fix(VELOCITY_X);
             it_node->Fix(VELOCITY_Y);
             it_node->Fix(VELOCITY_Z);
+
+            // We store previous info in order to reset after PFEM Solve
+            if (r_process_info[STEP] > 1) { // We have the previous acc
+                auto current_acceleration = it_node->FastGetSolutionStepValue(ACCELERATION, 0);
+                auto prev_acceleration    = it_node->FastGetSolutionStepValue(ACCELERATION, 1);
+                auto &r_current_accel_backup = it_node->FastGetSolutionStepValue(ACCELERATION_BACKUP);
+                auto &r_prev_accel_backup    = it_node->FastGetSolutionStepValue(ACCELERATION_PREVIOUS_BACKUP);
+                r_current_accel_backup = current_acceleration;
+                r_prev_accel_backup    = prev_acceleration;
+
+                auto current_displ = it_node->FastGetSolutionStepValue(DISPLACEMENT, 0);
+                auto prev_displ    = it_node->FastGetSolutionStepValue(DISPLACEMENT, 1);
+                auto &r_current_displ_backup = it_node->FastGetSolutionStepValue(DISPLACEMENT_BACKUP);
+                auto &r_prev_displ_backup    = it_node->FastGetSolutionStepValue(DISPLACEMENT_PREVIOUS_BACKUP);
+                r_current_displ_backup = current_displ;
+                r_prev_displ_backup    = prev_displ;
+            }
+
         } else {
             it_node->Free(VELOCITY_X);
             it_node->Free(VELOCITY_Y);
             it_node->Free(VELOCITY_Z);
+
+            // We store previous info in order to reset after PFEM Solve
+            if (r_process_info[STEP] > 1) { // We have the previous acc
+                auto &r_current_acceleration = it_node->FastGetSolutionStepValue(ACCELERATION, 0);
+                auto &r_prev_acceleration    = it_node->FastGetSolutionStepValue(ACCELERATION, 1);
+                auto current_accel_backup = it_node->FastGetSolutionStepValue(ACCELERATION_BACKUP);
+                auto prev_accel_backup    = it_node->FastGetSolutionStepValue(ACCELERATION_PREVIOUS_BACKUP);
+                r_current_acceleration = current_accel_backup;
+                r_prev_acceleration = prev_accel_backup;
+
+                auto &r_current_displ = it_node->FastGetSolutionStepValue(DISPLACEMENT, 0);
+                auto &r_prev_displ    = it_node->FastGetSolutionStepValue(DISPLACEMENT, 1);
+                auto current_displ_backup = it_node->FastGetSolutionStepValue(DISPLACEMENT_BACKUP);
```

Comment:
This backups take a lot of memory, and why do you need to have to variables for backup?, all historical variables have buffer, you can acces from previous positions